front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldItemTableRow.tsx

```diff
@@ -32,27 +36,50 @@ const StyledIconTableCell = styled(TableCell)`
 
 export const SettingsObjectFieldItemTableRow = ({
   ActionIcon,
-  fieldItem,
+  fieldMetadataItem: fieldMetadataItem,
 }: SettingsObjectFieldItemTableRowProps) => {
   const theme = useTheme();
-  const { Icon } = useLazyLoadIcon(fieldItem.icon ?? '');
+  const { Icon } = useLazyLoadIcon(fieldMetadataItem.icon ?? '');
+  const navigate = useNavigate();
 
   // TODO: parse with zod and merge types with FieldType (create a subset of FieldType for example)
-  const fieldDataTypeIsSupported = Object.keys(dataTypes).includes(
-    fieldItem.type,
-  );
+  const fieldDataTypeIsSupported = fieldMetadataItem.type in dataTypes;
```

Comment:
I think dataTypes should be renamed to SettingsFieldMetadataTypes

Reply:
Done